examples/interaction.rs

```diff
@@ -0,0 +1,292 @@
+                Self::Msg::CounterReplyCount(n) => {
+                    if n >= state.threshold {
+                        (
+                            SupervisorState {
+                                success: true,
+                                ..**state
+                            },
+                            Vec::new(),
+                        )
+                    } else {
+                        (**state, Vec::new())
+                    }
+                }
+
+                _ => (**state, Vec::new()),
+            };
+
+            *state.to_mut() = new_state;
+
+            for (addr, msg) in output_msgs {
+                o.send(addr, msg)
+            }
+        }
+    }
+}
+
+pub struct ExternalInput {
+    pub threshold: CounterSize,
+    pub supervisor_addr: Id,
+}
+
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub struct InputState {
+    pub wait_cycles: u32, // Only used for observing system evolution in the explorer.
+    pub done: bool,
+}
+
+// Timers are discrete and time out immediately for model checking purposes.
+// To trigger actions in certain orders they need to be set by the correct events.
+#[derive(Clone, Debug, Eq, Hash, PartialEq, Serialize, Deserialize)]
+pub enum InputTimer {
+    IncrementInput,
+    ReportInput,
+}
+
+impl Actor for ExternalInput {
+    type Msg = Msg;
+    type State = InputState;
+    type Timer = InputTimer;
+
+    fn on_start(&self, _id: Id, o: &mut Out<Self>) -> Self::State {
+        // Set a timeout to trigger sending increment request.
+        o.set_timer(InputTimer::IncrementInput, model_timeout());
+        InputState {
+            wait_cycles: 0,
+            done: false,
+        }
+    }
+
+    fn on_msg(
+        &self,
+        _id: Id,
+        state: &mut Cow<Self::State>,
+        _src: Id,
+        msg: Self::Msg,
+        _o: &mut Out<Self>,
+    ) {
+        match msg {
+            Msg::CounterReplyCount(n) => {
+                if n >= self.threshold {
+                    state.to_mut().done = true;
+                }
+            }
+            _ => (),
+        }
+    }
+
+    fn on_timeout(
+        &self,
+        _id: Id,
+        state: &mut Cow<Self::State>,
+        timer: &Self::Timer,
+        o: &mut Out<Self>,
+    ) {
+        // We use the reuse the message types of SupervisorMachine here to simulate a user triggering Supervisor behavior.
+        match timer {
+            InputTimer::IncrementInput => {
+                // Set timeout for requesting success status s.t. it happens after incrementing.
+                o.set_timer(InputTimer::ReportInput, model_timeout());
+                o.send(self.supervisor_addr, Msg::SupervisorIncrementRequest(3));
+                state.to_mut().wait_cycles += 1; // Increment InputCycles
+            }
+            InputTimer::ReportInput => {
+                o.send(self.supervisor_addr, Msg::SupervisorReportRequest());
+                state.to_mut().wait_cycles += 1;
+            }
```

Comment:
I don't think anything is generating these timer events.

Reply:
If I understand it correctly, you can chain timeouts, which is what I am doing here.

The initial one is set in `on_start` of this Actor, the following ones get set every time one timeout event happens. If I read the output of the stateright explorer correctly, this is also what happens during execution.